aws/rust-runtime/aws-config/src/lib.rs

```diff
@@ -540,6 +567,18 @@ mod loader {
                 ))
             };
 
+            let use_fips = if let Some(use_fips) = self.use_fips {
+                Some(use_fips)
+            } else {
+                use_fips_provider(&conf).await
+            };
+
+            let use_dual_stack = if let Some(use_dual_stack) = self.use_dual_stack {
+                Some(use_dual_stack)
+            } else {
+                use_dual_stack_provider(&conf).await
+            };
+
```

Comment:
I wish there was a better way to handle futures containing options but it looks like `TryFutureExt` only covers `Result`s. I don't like having two `=` on one line but the following alternative doesn't actually work so 🤷‍♀️.

```rust
async fn foo() {
    let config = Config::new();

    let use_dual_stack = future::ready(config.use_dual_stack)
        .or_else(|| use_dual_stack_provider(&config))
        .await;
    
    println!("use_dual_stack == {use_dual_stack:?}");
}
```

Reply:
I was considering writing a `or_else_fut` trait but figured it wasn't worth it

Reply:
Yeah, I agree with you.